i6_models/parts/conformer/frontend.py

```diff
@@ -0,0 +1,282 @@
+
+        self.conv1 = nn.Conv2d(
+            in_channels=model_cfg.features,
+            out_channels=model_cfg.channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.conv2 = nn.Conv2d(
+            in_channels=model_cfg.channels,
+            out_channels=model_cfg.channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.pooling = nn.MaxPool2d(
+            kernel_size=model_cfg.pool_kernel_size,
+            stride=model_cfg.pool_strides,
+            padding=pool_padding,
+        )
+        self.conv3 = nn.Conv2d(
+            in_channels=model_cfg.channels,
+            out_channels=model_cfg.channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.conv4 = nn.Conv2d(
+            in_channels=model_cfg.channels,
+            out_channels=model_cfg.channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.activation = model_cfg.activation
+
+    def forward(self, tensor: torch.Tensor) -> torch.Tensor:
+        """
+        T might be reduced to T' depending on ...
+
+        :param tensor: input tensor of shape [B,T,F]
+        :return: torch.Tensor of shape [B,T',F]
+        """
+        # conv layers expect shape [B,F,T] so we have to transpose here
+        tensor = torch.transpose(tensor, 1, 2)  # [B,F,T]
+        tensor = tensor[:, :, :, None]  # [B,F,T,C]
+        tensor = self.conv1(tensor)
+        tensor = torch.transpose(tensor, 1, 2)  # transpose back to [B,T,F,C]
+        tensor = torch.squeeze(tensor)  # [B,T,F]
+
+        tensor = self.activation(tensor)
+        tensor = self.pooling(tensor)
+
+        tensor = torch.transpose(tensor, 1, 2)  # [B,F,T]
+        tensor = tensor[:, :, :, None]  # [B,F,T,C]
+        tensor = self.conv2(tensor)
+        tensor = torch.transpose(tensor, 1, 2)  # transpose back to [B,T,F,C]
+        tensor = torch.squeeze(tensor)  # [B,T,F]
+
+        tensor = self.activation(tensor)
+
+        tensor = torch.transpose(tensor, 1, 2)  # [B,F,T]
+        tensor = tensor[:, :, :, None]  # [B,F,T,C]
+        tensor = self.conv3(tensor)
+        tensor = torch.transpose(tensor, 1, 2)  # transpose back to [B,T,F,C]
+        tensor = torch.squeeze(tensor)  # [B,T,F]
+
+        tensor = self.activation(tensor)
+
+        tensor = torch.transpose(tensor, 1, 2)  # [B,F,T]
+        tensor = tensor[:, :, :, None]  # [B,F,T,C]
+        tensor = self.conv4(tensor)
+        tensor = torch.transpose(tensor, 1, 2)  # transpose back to [B,T,F,C]
+        tensor = torch.squeeze(tensor)  # [B,T,F]
+
+        return tensor
+
+    @staticmethod
+    def _get_padding(input_size: Union[int, Tuple[int, ...]]) -> int:
+        return get_padding(input_size)
+
+
+def get_padding(input_size: Union[int, Tuple[int, ...]]) -> int:
```

Comment:
Maybe it should be private:
```suggestion
def _get_padding(input_size: Union[int, Tuple[int, ...]]) -> int:
```